.github/workflows/deploy-docs.yml

```diff
@@ -0,0 +1,70 @@
+name: Deploy docs on gh-pages
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build-environment:
+    runs-on: ubuntu-18.04
+    name: Build conda environment
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout repository
+
+      # See: https://github.com/marketplace/actions/setup-conda
+      - uses: s-weigand/setup-conda@v1
+        with:
+          conda-channels: "conda-forge"
+
+      # Build cache of environment -- this can shave minutes off the CI.
+      - name: Cache conda environment
+        id: cache-environment
+        uses: actions/cache@v2
+        # Conda environment build step depends on just environment.yml,
+        # so we ensure that the hash key contains its hash.
+        # If the file changes, then its hash will change,
+        # and the cache will be invalidated,
+        # thus triggering a rebuild.
+        # (There is a strong assumption here that changing `build_environment.sh`
+        # will not change the environment definition, which it shouldn't.)
+        with:
+          path: |
+            jax-unirep.tar.gz
+          key: ${{ runner.os }}-${{ hashFiles('environment.yml') }}
+
+      - name: Build and pack environment
+        if: steps.cache-environment.outputs.cache-hit != 'true'
+        run: bash scripts/ci/build_environment.sh
+
+      # See: https://github.com/actions/upload-artifact
+      - name: Upload environment
+        uses: actions/upload-artifact@v2
+        with:
+          name: jax-unirep-tarball
+          path: jax-unirep.tar.gz
+
+  deploy:
+    name: Deploy docs
+    needs: build-environment
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # https://github.com/actions/download-artifact
+      - name: Download environment tarball
+        uses: actions/download-artifact@v2
+        with:
+          name: jax-unirep-tarball
+
+      - name: Unpack environment and activate it
+        run: bash scripts/ci/unpack_environment.sh
+
+      - name: Deploy docs
```

Comment:
That's a cool find! I think, however, it might not work because the docs depend on the source code being in the environment. I _think_ what will happen here is that the docs build will fail, because the environment is missing.

I think you might want to:

1. Copy the `build-docs` step from the other CI pipeline,
2. Use the `peaceiris/actions-gh-pages@v3` action. I have an example [here](https://github.com/ericmjl/Network-Analysis-Made-Simple/blob/master/.github/workflows/deploy.yaml#L66).

[This other action](https://github.com/marketplace/actions/deploy-to-github-pages) also seems pretty good, but I think the peaceiris one is a bit more popular and has more options to explore.

Reply:
Changed it, secret already added as well.